pages/software/building.md

```diff
@@ -1,5 +1,157 @@
+popular software requests first so it's important you vote.
+
+## About EasyBuild
+
+CCR uses [EasyBuild](https://easybuild.io/) for building software. EasyBuild is
+a software build and installation framework that helps manage software on High
+Performance Computing (HPC) systems in an efficient and reproducible way.
+
+EasyBuild is written in python and uses simple human readable build recipes.
+EasyBuild provides a vast repository of build recipes for all kinds of software
+shared with the greater HPC community. There's a good chance that the software
+you're looking for already has an EasyBuild recipe.
+
+EasyBuild recipes, called easyconfigs, are text files (python syntax) used
+instruct EasyBuild how to build software. You can browse the community
+supported [recipes here](https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs).
+
+The standard EasyBuild recipe filename format is:
+
+```
+<software name>-<version>-<compiler>-<compiler version>.eb
+
+Example:
+
+  SAMtools-1.16.1-GCC-11.2.0.eb
+
+   Software Name: SAMtools
+         Version: 1.16.1
+        Compiler: GCC
+Compiler Version: 11.2.0
+```
+
+When building software with EasyBuild, it's very important to pay attention to
+the compiler and compiler version as it will be much easier to build software
+for [a compiler CCR already supports](releases.md).
+
+For more information about EasyBuild [see here](https://docs.easybuild.io/en/latest/).
+
+EasyBuild will take care of compiling, installing, and creating a Modulefile
+for you. CCR provides EasyBuild as it's own module and has pre-configured
+EasyBuild with the appropriate settings for CCR's environment. To load CCR's
+easybuild module simply run this:
+
+```
+$ module load easybuild
+```
+
+!!! Note 
+    Compiling certain software will be CPU architecture specific. If you'd like
+    to build software for a specific CPU architecture, be sure to load the
+    corresponding `ccrarch/avx2` or `ccrarch/avx512` module first.
+
+## Building a new software module
+
+If you can't find the software you're looking for from the [available software](modules.md) 
+provided by CCR, you can attempt to build it yourself. You can use easybuild to
+search for recipes to build. For example:
+
+```
+$ module load easybuild
+$ eb -S some_package_name
+
+# For example
+$ eb -S samtools
+ * $CFGS1/s/SAMtools/SAMtools-1.13-GCC-10.3.0.eb
+ * $CFGS1/s/SAMtools/SAMtools-1.13-GCC-11.3.0.eb
+ * $CFGS1/s/SAMtools/SAMtools-1.14-GCC-11.2.0.eb
+ * $CFGS1/s/SAMtools/SAMtools-1.15-GCC-11.2.0.eb
+ * $CFGS1/s/SAMtools/SAMtools-1.15.1-GCC-11.2.0.eb
+ * $CFGS1/s/SAMtools/SAMtools-1.16.1-GCC-11.2.0.eb
+```
+
+Select the recipe, be sure to select the one [for a supported compiler
+toolchain](releases.md) if necessary and build the software by running the
+following command:
+
+```
+# Build software
+$ eb SAMtools-X.X.X-GCC-Y.Y.Y.eb
+
+# Check what are the dependencies
+$ eb SAMtools-X.X.X-GCC-Y.Y.Y.eb -M
+
```

Comment:
Would recommend switching this order.  Show command for checking the dependencies first, then the command to build the software, then the command to build sw + dependencies.